### PR TITLE
Use AMP Geo to hide AMP ads

### LINKED ISSUE #6795: Use amp-geo in amp-ads
**Is your feature request related to a problem? Please describe.**
We need to decide how to ensure we only show advertisements to users outside of the uk on our AMP pages. One solution to be considered is https://amp.dev/documentation/components/amp-geo/. This is already used in Simorgh to decide when to show cookie banners. We need to determine the accuracy of this, and how it compares to https://www.bbc.com/userinfo/.

We are no longer using the geoip information from the toggles service in the AMP ads implementation. For this investigation, we should use amp-geo to hide adverts inside the uk, and put this onto the test environment.

This can then be used to help product/GNL decide if the data we receive here is accurate enough to make decisions around showing advertisements.

It is known that this will not contain the corrections or proxy detection of the BBC GeoIP service.

**Describe the solution you'd like**

- update [psammead-amp-geo](https://github.com/bbc/psammead/blob/latest/packages/components/psammead-amp-geo/src/index.jsx) to have a separete group (for ad locations - non-UK)
  - What counts as non-UK? According to GeoIP headers confluence page (`/display/geoip/GeoIP+Headers`), GeoIP will resolve all GB country codes (i.e. 'gb', 'uk', 'gg' (Guernsey), 'im' (Isle of Man), 'je' (Jersey)) to X-Country = 'gb'. We should also not show ads if the country code cannot be determined due to connection issues.
- Use amp-geo in amp-ads
- ensure default is to not show ads (simorgh ads container).

**Describe alternatives you've considered**

**Testing notes**
We would like GNL to also test this.

On our side, we can use VPNs, proxies and get our non-UK engineers to check.

We should perform a regression on the Cookie Banner to ensure that changes to psammead-amp-geo have not caused any changes in behaviour.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**
Guide for UK BBC devs to access proxy: https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/testing-ads.md

### PR DESCRIPTION
**Pull request by tochwill**
Resolves #6795

**Overall change:**
This is reliant on the update in https://github.com/bbc/psammead/pull/3539/.

This uses AMP Geo to show or hide AMP ads based on whether the user is located within the UK.

**Code changes:**

- Adds an `.amp-geo-pending` class to the body of the AMP document. This is required to delay rendering of components until AMP Geo information is available.
- Hides AMP ads if geolocation data is not yet available, or the user is within the 'gbOrUnknown' geolocation group.
- Setting the AMP ad container to display: none hides the ad, and also stops any network requests being made (as outlined here: https://amp.dev/documentation/components/amp-geo/#render-blocking)
---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added labels to this PR for the relevant pod(s) affected by these changes
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

This can be tested both inside and outside the UK using a proxy - inside the UK we should see an ad, outside the UK we should not.
We should also ensure the cookie banner continues to behave as expected with the updates to `psammead-amp-geo`.

### CODE REVIEW

File: src/app/containers/Ad/Amp/index.jsx
Comment (tochwill) [inline src/app/containers/Ad/Amp/index.jsx:32]: Setting this visibility here means that we will see content shifting when the geolocation data becomes available. There is a discussion on the ads paper document to decide on how we want to approach this.

File: src/server/Document/component.jsx
Comment (tochwill) [inline src/server/Document/component.jsx:80]: We may want to add a cypress test for ads to ensure the AMP component is removing this and loading the ad in.

**Comment by tochwill** [comment]
As ad toggles are all currently set to false, AMP ads will not currently render. We should fix that as part of https://github.com/bbc/simorgh/issues/6866 - I don't think we can test this until then, so I'll block this.

**Comment by tochwill** [comment]
Blocked on https://github.com/bbc/simorgh/issues/6866 (or toggling ads back on locally/on test)

File: src/server/Document/component.jsx
Comment (tochwill) [inline src/server/Document/component.jsx:80]: I think the only way to test that the pending class is removed by the AMP Geo component and the ads display once that is returned is to actually run this in a browser (using Cypress). We could add this as a test for a page with display advertisement.

Comment (itimirichard) [reply src/server/Document/component.jsx:80]: Sounds like a good idea

**Comment by tochwill** [comment]
I was hoping we could have an amp-geo component contained within each component that needs to use it, but we can only have one per page. Will refactor this slightly.

**Comment by tochwill** [comment]
I'm resolving these conflicts, just making sure that our markup remains semantically correct in all the different scenarios.

**Review by thekp — APPROVED** [review]
LGTM, 

With local toggles set to true, I checked via proxy to the US and I can see Ads and when I was in the UK I can not see Ads. (Also it's a valid AMP page ✅ )
image

File: src/app/containers/Ad/Amp/index.jsx
Comment (DenisHdz) [inline src/app/containers/Ad/Amp/index.jsx:99]: In the PR description says that the `psammead-amp-geo` has been added into this container, but can't find it. Is that still in progress or I might have missed something?

File: src/app/containers/Ad/Amp/index.jsx
Comment (DenisHdz) [inline src/app/containers/Ad/Amp/index.jsx:33]: Where are the geo groups specified and where is this class added by AMP?

File: src/app/containers/Ad/Amp/index.jsx
Comment (tochwill) [reply src/app/containers/Ad/Amp/index.jsx:99]: Sorry, I have updated that. We have to reuse the amp-geo component that is being used by the rest of the components in Simorgh, as you can't have more than one on a page.

File: src/app/containers/Ad/Amp/index.jsx
Comment (tochwill) [reply src/app/containers/Ad/Amp/index.jsx:33]: These are added by https://github.com/bbc/psammead/blob/latest/packages/components/psammead-amp-geo/src/index.jsx. `amp-geo` adds and removes classes to the body of the document: https://amp.dev/documentation/components/amp-geo - I have linked to the relevant section elsewhere, but maybe it's worth explaining in a comment here too. I think it's quite difficult to understand this behaviour unless you've read up on how the component works, so I agree it could use some explanation.

File: src/app/containers/Ad/Amp/index.jsx
Comment (DenisHdz) [reply src/app/containers/Ad/Amp/index.jsx:99]: Ahh ok 👍

File: src/app/containers/Ad/Amp/index.jsx
Comment (DenisHdz) [reply src/app/containers/Ad/Amp/index.jsx:22]: Yeah, I saw you linked it in https://github.com/bbc/simorgh/pull/6961/files#diff-3e335d5559a68f6c5fd831d451b1ce70R44, but it would great to also explain here how AMP will remove the `amp-geo-pending` class and add the `amp-geo-group-` classes to hide the ads for those groups.

File: package-lock.json
Comment (tochwill) [inline package-lock.json]: hmmmmmmm

File: src/app/containers/Ad/__snapshots__/index.test.jsx.snap
Comment (tochwill) [inline src/app/containers/Ad/__snapshots__/index.test.jsx.snap]: I think styled components is stripping these out, have seen this before.

Comment (tochwill) [reply src/app/containers/Ad/__snapshots__/index.test.jsx.snap]: Fixing now!

**Review by DenisHdz — APPROVED** [review]
Thanks for the latest changes and comments 👍 

amp-geo

**Comment by PriyaKR** [comment]
Testing:

http://localhost:7080/arabic.amp?adUnit=/4817/bbccom.test.site.flash/news_homepage_int

With the ads enabled for arabic in isite and when in uk the ads are not displaced on the amp page but when set to US proxy ads are seen.
Also @itimirichard confirmed he is able to see ads outside UK.
Regression testing on amp cookie banners looks good.Outisde EEA the cookie banners are not displayed and inside EEA the banners are displayed.

This PR can be merged once when this use case behaviour is confirmed - when the access to the isite config is blocked(both server and client side) also i have set the proxy to non uk location  and the hard coded config ads value is set to true in simorgh for local env canonical seems to respect this value and displays the ad but the amp page doesn't respect this value and does not displays the ads.Is that the expected behaviour for amp pages?

**Comment by PriyaKR** [comment]
> Testing:
> 
> http://localhost:7080/arabic.amp?adUnit=/4817/bbccom.test.site.flash/news_homepage_int
> 
> With the ads enabled for arabic in isite and when in uk the ads are not displaced on the amp page but when set to US proxy ads are seen.
> Also @itimirichard confirmed he is able to see ads outside UK.
> Regression testing on amp cookie banners looks good.Outisde EEA the cookie banners are not displayed and inside EEA the banners are displayed.
> 
> This PR can be merged once when this use case behaviour is confirmed - when the access to the isite config is blocked(both server and client side) also i have set the proxy to non uk location and the hard coded config ads value is set to true in simorgh for local env canonical seems to respect this value and displays the ad but the amp page doesn't respect this value and does not displays the ads.Is that the expected behaviour for amp pages?

@tochwill as discussed with Tom it seems to be the expected behaviour for amp so this PR can be merged now.

**Pull request merged**
